secrets/vault/vault_test.go

```diff
@@ -21,19 +21,19 @@ import (
 	"testing"
 
 	"github.com/hashicorp/vault/api"
-	"github.com/hashicorp/vault/builtin/logical/transit"
-	vhttp "github.com/hashicorp/vault/http"
-	"github.com/hashicorp/vault/logical"
-	"github.com/hashicorp/vault/vault"
 	"gocloud.dev/secrets"
 	"gocloud.dev/secrets/driver"
 	"gocloud.dev/secrets/drivertest"
 )
 
+// To run these tests against a real Vault server, first run ./localvault.sh.
+// Then wait a few seconds for the server to be ready.
```

Comment:
Is there a way we can health check it for readiness? Or perhaps dial-looping for some period of time? This seems like a source of flakiness.

Reply:
Gotcha. I do wonder whether start_local_deps.sh or localvault.sh should wait for a healthy response before continuing with the tests.

Reply:
Yeah I thought about doing some checks on the server health and retries. But I think it is hard to  know whether we should restart the container or wait longer for the health probe. It would become as flaky. That's why I did more retries when dialing the test server and skip the test after a few attempts. WDYT?

Reply:
What I'm saying is: the test is using presence of server to determine whether to run the test, so that should fail fast. The localvault.sh script should block on health checks before coming up. This is likely a problem with all current local dep scripts, so it would be fine to file an issue and handle it separately. In the meantime, I think these tests should fail fast.